core/src/main/java/amino/run/oms/migrationdecision/MetricWatcher.java

```diff
@@ -0,0 +1,282 @@
+           tk = Tk,posk+i + summation { (Ski+Sik)/Rposk,posi + Lposk,posi }
+        */
+
+        // get current kernel server for replica
+        InetSocketAddress replicaKernelServer =
+                microServiceManager.getReplicaDispatcher(replicaID).getHost();
+        // get metric stat for current replica kernel server
+        Map<InetSocketAddress, NodeMetric> replicaNodeMetric =
+                serverManager.getKernelServerMetric(replicaKernelServer);
+
+        // get metrics stat of RPC processing time from different kernel client
+        // TODO currently considering processing time from different kernel client for simplicity.
+        //  Need to get processing time within micro service replica
+        Map<InetSocketAddress, RPCMetric> replicaMetrics =
+                microServiceManager.getMicroServiceMetric(replicaID);
+
+        // calculate time factor impact on RPC with latency and bandwidth
+        long averageExecutionTime = 0;
+        long summation = 0;
+        long data;
+        NodeMetric nodeMetric;
+        for (Map.Entry<InetSocketAddress, RPCMetric> replicaMetric : replicaMetrics.entrySet()) {
+            averageExecutionTime += replicaMetric.getValue().elapsedTime;
+
+            data = replicaMetric.getValue().dataSize;
+            nodeMetric = replicaNodeMetric.get(replicaMetric.getKey());
+            // kernel client utilized by micro service but not running in kernel server are
+            // ignored in process time computation because no metric data available of kernel server
+            // from there.
+            if (nodeMetric == null) {
+                logger.info("skipping kernel client not part of kernel server");
+                continue;
+            }
+
+            summation += (data / nodeMetric.rate + nodeMetric.latency);
+        }
+
+        if (!replicaMetrics.isEmpty()) {
+            averageExecutionTime = averageExecutionTime / replicaMetrics.size();
+        }
+
+        return averageExecutionTime + summation;
+    }
+
+    private List<ReplicaID> getMigrationReplicas() {
+        List<ReplicaID> replicaIDs = new ArrayList<ReplicaID>();
+
+        try {
+            List<MicroServiceID> microServiceIDS = microServiceManager.getAllMicroServices();
+            // TODO Assuming all replica can migrate. In future, user can
+            //  restrict replica migration to specific set of replica.
+            for (MicroServiceID microServiceID : microServiceIDS) {
+                try {
+                    replicaIDs.addAll(microServiceManager.getReplicaIDs(microServiceID));
+                } catch (MicroServiceNotFoundException e) {
+                    logger.warning(
+                            String.format(
+                                    "MicroService [%s] not available. Skipping replicas in migration evaluation",
+                                    microServiceID));
+                    e.printStackTrace();
```

Comment:
What is the purpose of printing stack traces all over the place like this?  It makes it very difficult to look at the console output and tell whether anything has gone wrong or not.